### ISSUE TITLE
Armv7 related crash on build using SPM

### ISSUE THREAD

**Issue by Umar-M-Haroon** [firebase/firebase-ios-sdk#7235]
### Step 0: Are you in the right place?

  * For issues or feature requests related to __the code in this repository__
    file a Github issue.
    * If this is a __feature request__ please use the Feature Request template.
  * For general technical questions, post a question on [StackOverflow](http://stackoverflow.com/)
    with the `firebase` tag.
  * For general (non-iOS) Firebase discussion, use the [firebase-talk](https://groups.google.com/forum/#!forum/firebase-talk)
    google group.
  * For backend issues, console issues, and other non-SDK help that does not fall under one
    of the above categories, reach out to
    [Firebase Support](https://firebase.google.com/support/).
  * Once you've read this section and determined that your issue is appropriate for
    this repository, please delete this section.

### [REQUIRED] Step 1: Describe your environment

  * Xcode version: 12.3
  * Firebase SDK version: 7.31.0
  * Installation method: `Swift Package Manager` (select one)
  * Firebase Component: Firestore (Auth, Core, Database, Firestore, Messaging, Storage, etc)

#### Steps to reproduce:

What happened? How can we make the problem occur?
This could be a description, log/console output, etc.

If you have a downloadable sample project that reproduces the bug you're reporting, you will
likely receive a faster response on your issue.

#### Relevant Code:
Hi! Sorry for the vagueness here, I'm trying to look towards a more reproducible method.

In short, I was working on archiving my app, which uses SPM and has Firestore, Dynamic Links, Messaging, Auth, and Storage onto a framework target. The main app uses the framework and usually runs fine. As I compiled for MacCatalyst, I had no issues, but when I compiled for any iOS device, I got the error 

`scattered reloc r_address too large for architecture armv7`

I've tried setting armv7 as an excluded arch everywhere, but no avail. I'm wondering why it's even compiling for armv7 though. My deployment target is iOS 13.0+ so shouldn't it only need arm64? I found an old tweet regarding a similar bug, but for i386 that recommended a refactor due to the project being too large, but their project was about 100x bigger than mine.

**Comment by paulb777** [comment]
From googling the error, it looks like a long-standing Xcode issue related to the size of the app.

As to why, Xcode is building armv7 when the deployment target is iOS 13, that seems like an Xcode/SPM bug. There's no reason to build and link dependencies arm7 if the app has something higher, even if the dependencies support iOS 10.

Perhaps worth asking at https://forums.swift.org/c/development/SwiftPM/ ?

I'll leave this open for discussion for awhile, but this is unlikely to be a Firebase issue.

**Comment by Umar-M-Haroon** [comment]
Interesting, I'll definitely post there and hope to get a better result. I'm hoping it's not some weird size bug, as my app is only 10K LOC

**Comment by Umar-M-Haroon** [comment]
here's an update, if I connect my device, select that as a run destination, then hit archive, then it succeeds.

**Comment by paulb777** [comment]
https://forums.swift.org/t/using-firebase-via-spm-facing-an-armv7-error-on-archive/43476

**Comment by paulb777** [comment]
We'll likely need a reproducible example to provide additional assistance here.
